app/services/my_anime_list_sync_service.rb

```diff
@@ -0,0 +1,203 @@
+      )
+    else
+      # find the anime or manga
+      get("#{media_type}/#{mal_media_id}#{MINE}", profile) do |response|
+        # need to check watched_status OR read_status
+        if media_type == 'anime'
+          if response['watched_status']
+            put(
+              "animelist/#{media_type}/#{mal_media_id}",
+              profile,
+              {
+                status: format_status(le.status),
+                episodes: le.progress,
+                score: format_score(le.rating),
+                rewatch_count: le.reconsume_count
+              }
+            )
+          else
+            post(
+              "animelist/#{media_type}",
+              profile,
+              {
+                anime_id: mal_media_id,
+                status: format_status(le.status),
+                episodes: le.progress,
+                score: format_score(le.rating)
+              }
+            )
+          end
+        else
+          # manga
+          if response['read_status']
+            put(
+              "mangalist/#{media_type}/#{mal_media_id}",
+              profile,
+              {
+                status: format_status(le.status),
+                chapters: le.progress,
+                score: format_score(le.rating),
+                reread_count: le.reconsume_count
+              }
+            )
+          else
+            post(
+              "mangalist/#{media_type}",
+              profile,
+              {
+                manga_id: mal_media_id,
+                status: format_status(le.status),
+                chapters: le.progress,
+                score: format_score(le.rating)
+              }
+            )
+          end
+        end
+      end
+
+      # if block is not executed because
+      # response.code was not a 200
+      # TODO: check timing
+      # return nil
+    end
+  end
+
+  def format_status(status)
+    # change our status -> mal status
+    case status
+    when 'current' then 1 # watching/reading
+    when 'planned' then 6 # plan to watch/plan to read
+    when 'completed' then 2 # completed
+    when 'on_hold' then 3 # on hold
+    when 'dropped' then 4 # dropped
+    end
+  end
+
+  # if you send no score in
+  # ie: &score&anythingelse
+  # it will not set the score
+  def format_score(score)
+    return nil if score.nil?
+
+    (score * 2).floor
+  end
+
+  private
+
+  def get(url, profile)
+    request = Typhoeus::Request.new(
+      build_url(url),
+      method: :get,
+      userpwd: simple_auth(profile)
+    )
+
+    request_status(request) do |response|
+      yield response
+    end
+    request.run
+  end
+
+  def post(url, profile, body)
+    request = Typhoeus::Request.new(
+      build_url(url),
+      method: :post,
+      userpwd: simple_auth(profile),
+      body: body
+    )
+
+    # TODO: @nuck is there a better way to handle this?
+    request_status(request) do |response|
+      # doesn't need to do anything if success
+    end
+    request.run
```

Comment:
I'd personally recommend just using Typheous in synchronous mode. If you call `Typhoeus::Request.new('https://example.com').run`, it returns a `Typheous::Response`.

You can call the `response_body`, `response_headers`, or `response_code` methods on the `Typheous::Response` to get info about the response